executor/modes/local/websocket_client.py

```
@@ -486,12 +478,17 @@ async def send_heartbeat(self, timeout: float = 5.0) -> bool:
         """Disconnect from the WebSocket server."""
-        if self._connected:
-            try:
-                await self.sio.disconnect()
+        try:
+            # Always attempt disconnect to ensure Engine.IO monitor threads are stopped,
+            # even if we never reached the "connected" state.
+            await self.sio.disconnect()
+            if self._connected:
                 logger.info("WebSocket disconnected gracefully")
-            except Exception as e:
+        except Exception as e:
```

Ruff (0.14.14)

[warning] Do not catch blind exception: `Exception`

(BLE001)